src/include/switch_caller.h

```diff
@@ -185,6 +185,20 @@ SWITCH_DECLARE(void) switch_caller_extension_add_application(_In_ switch_core_se
 															 _In_ switch_caller_extension_t *caller_extension,
 															 _In_z_ const char *application_name, _In_z_ const char *extra_data);
 
+/*!
+  \brief Insert an application (instruction) to the given extension, before any of the low priority applications
+  \param session session associated with the extension (bound by scope)
+  \param caller_extension extension to add the application to
+  \param application_name the name of the application
+  \param extra_data optional argument to the application
+  \param low_priority_applications array of application names to insert before
+  \param low_priority_count number of applications in the array
+*/
+SWITCH_DECLARE(void) switch_caller_extension_insert_application(_In_ switch_core_session_t *session,
```

Comment:
Is it possible to insert multiple application instead of one at a time? I think this would be efficient since we can only locate the preferred application once.

Reply:
i'm on it

Reply:
Should be completed. I also added a new test to check the algorithm.